src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingTask.cs

```diff
@@ -0,0 +1,108 @@
+        public bool VSTestBlameCrash { get; set; }
+        public string VSTestBlameCrashDumpType { get; set; }
+        public bool VSTestBlameCrashCollectAlways { get; set; }
+        public bool VSTestBlameHang { get; set; }
+        public string VSTestBlameHangDumpType { get; set; }
+        public string VSTestBlameHangTimeout { get; set; }
+        public ITaskItem VSTestTraceDataCollectorDirectoryPath { get; set; }
+        public bool VSTestNoLogo { get; set; }
+
+        public override bool Execute()
+        {
+            var traceEnabledValue = Environment.GetEnvironmentVariable("VSTEST_BUILD_TRACE");
+            Tracing.traceEnabled = !string.IsNullOrEmpty(traceEnabledValue) && traceEnabledValue.Equals("1", StringComparison.OrdinalIgnoreCase);
+
+            var debugEnabled = Environment.GetEnvironmentVariable("VSTEST_BUILD_DEBUG");
+            if (!string.IsNullOrEmpty(debugEnabled) && debugEnabled.Equals("1", StringComparison.Ordinal))
+            {
+                Console.WriteLine("Waiting for debugger attach...");
+
+                var currentProcess = Process.GetCurrentProcess();
+                Console.WriteLine(string.Format("Process Id: {0}, Name: {1}", currentProcess.Id, currentProcess.ProcessName));
+
+                while (!Debugger.IsAttached)
+                {
+                    Thread.Sleep(1000);
+                }
+
+                Debugger.Break();
+            }
+
+            // Avoid logging "Task returned false but did not log an error." on test failure, because we don't
+            // write MSBuild error. https://github.com/dotnet/msbuild/blob/51a1071f8871e0c93afbaf1b2ac2c9e59c7b6491/src/Framework/IBuildEngine7.cs#L12
+            var allowfailureWithoutError = BuildEngine.GetType().GetProperty("AllowFailureWithoutError");
+            allowfailureWithoutError?.SetValue(BuildEngine, true);
+
+            var processInfo = new ProcessStartInfo
+                                  {
+                                      FileName = toolExe,
+                                      Arguments = this.CreateCommandLineArguments(),
+                                      UseShellExecute = false,
+                                  };
+
+            if (!string.IsNullOrEmpty(this.VSTestFramework))
+            {
+                Console.WriteLine(Resources.TestRunningSummary, this.TestFileFullPath, this.VSTestFramework);
+            }
+
+            Tracing.Trace("VSTest: Starting vstest.console...");
+            Tracing.Trace("VSTest: Arguments: " + processInfo.FileName + " " + processInfo.Arguments);
+
+            using (var activeProcess = new Process { StartInfo = processInfo })
+            {
+                activeProcess.Start();
+                this.activeProcessId = activeProcess.Id;
+
+                activeProcess.WaitForExit();
+                Tracing.Trace("VSTest: Exit code: " + activeProcess.ExitCode);
+                return activeProcess.ExitCode == 0;
```

Comment:
We should only return `!Log.HasLoggedErrors`. The task should fail if and only if it has logged any errors.

Reply:
This is usually the way it's done in MSBuild tasks. Does `vstest.console.exe` return an error code even in cases of success?
By the way this task has been mostly left unchanged from current behaviour and kept for compatibility. The goal of this PR is rather to introduce a new task that would be more respectful of MSBuild, see `VSTestTask`.